documentation/istio_keycloak_integration.asciidoc

```diff
@@ -0,0 +1,213 @@
+:toc: macro
+toc::[]
+:idprefix:
+:idseparator: -
+== Authentication with Keycloak and Istio
+
+=== Introducing Keycloak
+https://www.keycloak.org/[Keycloak] is an opensource IAM (Identity and Access Management) solution with a broad set of features like SSO, authentication and authorization, social login, multifactor authentication etc.
+
+=== Introducing Istio
+https://istio.io/latest/docs/concepts/[Istio] is an opensource _https://istio.io/latest/about/service-mesh/[Service Mesh]_ that helps organizations run distributed MS-based apps anywhere.
+It contains both a control plane and a data plane (Envoy Sidecar). The control plane takes the desired configuration, and its view of the services, and dynamically programs the proxy servers, updating them as the rules or the environment changes. Its powerful features provide a uniform and more efficient way to secure, connect, and monitor services.
+
+* *Traffic management*: managing flow of traffic among services, from external and simplifying configuration of service-level properties like circuit breakers, timeouts etc.
+* *Observability*: generating detailed telemetry includes detailed metrics, distributed traces, and full access logs
+* *Security*: providing strong identity, powerful policy, transparent TLS encryption, and among-services/external/RBAC authentication, authorization and audit (AAA) tools
+
+== Setting up the Quarkus reference application with Istio and Keycloak
+
+This is a step-by-step guide to setting up the Quarkus reference application in conjunction with Istio and Keycloak. This setup was tested in a local Kubernetes cluster in a WSL environment. To set up the environment, follow these instructions: https://1000kit.gitlab.io/guides/docs/dev-environment/wsl2-pure/.
+
+=== Create the Kubernetes cluster
+To create the cluster, we use k3d. First create a local Docker registry.
+```
+k3d registry create registry --port 5000
+```
+The next step is to create the cluster. Navigate to the folder with the reference application and run the following command in your WSL environment.
+```
+k3d cluster create -c k8s/cluster-setup.yaml --k3s-server-arg '--no-deploy=traefik'
+```
+This will create a local Kubernetes cluster with one master and two worker nodes within your WSL environment. By default, k3d creates a Traefik proxy as an ingress controller. Since we will use the Istio Ingress Gateway later, we do not need it here.
```

Comment:
Shall we really commit ourselves to `k3d`?
In the PR for devonfw-ide I suggested that we go for docker desktop.
In real environments we will always use kubernetes natively.
So if we locally work with `k3d` the guide only works locally and only if `k3d` is present...

I am happy to be told otherwise but why dont we describe this directly with `kubectl`?